client/web/compose/src/components/ModuleFields/Editor/Geometry.vue

```diff
@@ -229,6 +244,43 @@ export default {
         this.localValue = { coordinates: [] }
       }
     },
+
+    currentLocation () {
+      try {
+        if (navigator.geolocation) {
+          navigator.geolocation.getCurrentPosition(
+            position => {
+              this.localValue = {
+                coordinates: [
+                  position.coords.latitude,
+                  position.coords.longitude,
+                ],
+              }
+            },
+            error => {
+              switch (error.code) {
+                case error.PERMISSION_DENIED:
+                  this.toastErrorHandler(this.$t('notification:field-geometry.geolocationErrors.permissionDenied'))()
+                  break
+                case error.POSITION_UNAVAILABLE:
+                  this.toastErrorHandler(this.$t('notification:field-geometry.geolocationErrors.positionUnavailable'))()
+                  break
+                case error.TIMEOUT:
+                  this.toastErrorHandler(this.$t('notification:field-geometry.geolocationErrors.timeout'))()
+                  break
+                default:
+                  this.toastErrorHandler(this.$t('notification:field-geometry.geolocationErrors.unknownError'))()
+                  break
+              }
+            },
+          )
+        } else {
```

Comment:
Move this to the top and to !navigator.geolocation instead and return inside it, so its less nesting